plugins/dell-kestrel/fu-dell-kestrel-ec-hid.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2024 Dell Technologies
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include <errno.h>
+#include <string.h>
+
+#include "fu-dell-kestrel-ec-hid.h"
+
+gboolean
+fu_dell_kestrel_ec_hid_write(FuDevice *device, GBytes *buf, GError **error)
+{
+	guint8 *data = (guint8 *)g_bytes_get_data(buf, NULL);
+	gsize data_sz = g_bytes_get_size(buf);
+
+	return fu_hid_device_set_report(FU_HID_DEVICE(device),
+					0x0,
+					data,
+					data_sz,
+					DELL_KESTREL_EC_HID_TIMEOUT,
+					FU_HID_DEVICE_FLAG_RETRY_FAILURE,
+					error);
+}
+
+GBytes *
+fu_dell_kestrel_ec_hid_fwup_pkg_new(FuChunk *chk,
+				    gsize fw_size,
+				    FuDellKestrelEcDevType dev_type,
+				    guint8 dev_identifier)
+{
+	g_autoptr(GByteArray) fwbuf = g_byte_array_new();
+	gsize chk_datasz = fu_chunk_get_data_sz(chk);
+
+	/* header */
+	fu_byte_array_append_uint8(fwbuf, DELL_KESTREL_EC_HID_CMD_FWUPDATE);
+	fu_byte_array_append_uint8(fwbuf, DELL_KESTREL_EC_HID_EXT_FWUPDATE);
+	fu_byte_array_append_uint32(fwbuf, 7 + chk_datasz, (G_BIG_ENDIAN)); // 7 = sizeof(command)
+
+	/* command */
+	fu_byte_array_append_uint8(fwbuf, DELL_KESTREL_EC_HID_SUBCMD_FWUPDATE);
+	fu_byte_array_append_uint8(fwbuf, dev_type);
+	fu_byte_array_append_uint8(fwbuf, dev_identifier);
+	fu_byte_array_append_uint32(fwbuf, fw_size, G_BIG_ENDIAN);
+
+	/* data */
+	fu_byte_array_append_bytes(fwbuf, fu_chunk_get_bytes(chk));
```

Comment:
can this be in rust format too? I think it would be a nice cleanup especially as these numbers are BE, not LE.

Reply:
the chunk size needs to be deretmined at runtime, unlikely to be predefined in rs.

Reply:
you can leave out the data -- the FuStruct is actually a GByteArray under the covers so `fu_byte_array_append_bytes(fwbuf, fu_chunk_get_bytes(chk));` will still work with the structure type!